lib/chef/resource/timezone.rb

```diff
@@ -48,90 +56,118 @@ class Timezone < Chef::Resource
         description: "An optional property to set the timezone value if it differs from the resource block's name.",
         name_property: true
 
-      action :set do
-        description "Set the timezone."
-
-        # some linux systems may be missing the timezone data
-        if linux?
-          package "tzdata" do
-            package_name suse? ? "timezone" : "tzdata"
-          end
+      # detect the current TZ on darwin hosts
+      #
+      # @since 14.7
+      # @return [String] TZ database value
+      def current_darwin_tz
+        tz_shellout = shell_out!(["systemsetup", "-gettimezone"])
+        if /You need administrator access/.match?(tz_shellout.stdout)
+          raise "The timezone resource requires administrative privileges to run on macOS hosts!"
+        else
+          /Time Zone: (.*)/.match(tz_shellout.stdout)[1]
         end
+      end
 
-        # Modern SUSE, Amazon, Fedora, RHEL, Ubuntu & Debian
-        if systemd?
-          cmd_set_tz = "/usr/bin/timedatectl --no-ask-password set-timezone #{new_resource.timezone}"
+      # detect the current timezone on windows hosts
+      #
+      # @since 14.7
+      # @return [String] timezone id
+      def current_windows_tz
+        tz_shellout = shell_out!("tzutil /g")
+        raise "There was an error running the tzutil command" if tz_shellout.exitstatus == 1
```

Comment:
There's a few `.exitstatus == 1` calls in here. Any reason to be so specific about the exit code instead of `.error?`?

Reply:
We have a mix historically. I switched this over to error? for reability